File: 1293-three-consecutive-odds/1293-three-consecutive-odds.java
class Solution {
    public boolean threeConsecutiveOdds(int[] arr) {
        int required = 0;

        for(int num : arr)
        {
            if(required == 3) 
                break; 
            if(num % 2 == 1){
                required++;
            }else 
                required = 0;
        }

        return required == 3 ? true:false;
    }
}

File: 1293-three-consecutive-odds/1293-three-consecutive-odds.kt
class Solution {
    fun threeConsecutiveOdds(arr: IntArray): Boolean {
        var oddCount = 0

        for(i in arr.indices){
            if(oddCount == 3) return true
            if(arr[i] % 2 != 0){
                oddCount++
            }else oddCount = 0
        }
        return if(oddCount == 3) true else false
    }
}